test.js

```diff
@@ -0,0 +1,193 @@
+        if (err) {
+          return next(err);
+        }
+        console.log('starting container from image: %s', imgName);
+        docker.createContainer({
+          Image: imgName,
+          HostConfig: {
+            PublishAllPorts: true,
+          },
+        }, function(err, container) {
+          console.log('recording container for later cleanup: ', container.id);
+          containerToDelete = container;
+          if (err) {
+            return next(err);
+          }
+          container.start(function(err, data) {
+            next(err, container);
+          });
+        });
+      });
+    });
+  };
+}
+
+function setCloudantEnv(container, next) {
+  container.inspect(function(err, c) {
+    // if swarm, Node.Ip will be set to actual node's IP
+    // if not swarm, but remote docker, use docker host's IP
+    // if local docker, use localhost
+    var host = _.get(c, 'Node.IP', _.get(docker, 'modem.host', '127.0.0.1'));
+    // container's port 80 is dynamically mapped to an external port
+    var port = _.get(c,
+      ['NetworkSettings', 'Ports', '80/tcp', '0', 'HostPort']);
+    process.env.CLOUDANT_PORT = port;
+    process.env.CLOUDANT_HOST = host;
+    process.env.CLOUDANT_URL = 'http://' + host + ':' + port;
+    console.log('env:', _.pick(process.env, [
+      'CLOUDANT_URL',
+      'CLOUDANT_HOST',
+      'CLOUDANT_PORT',
+      'CLOUDANT_USERNAME',
+      'CLOUDANT_PASSWORD',
+      'CLOUDANT_DATABASE',
+    ]));
+    next(null, container);
+  });
+}
+
+function waitFor(path) {
+  return function waitForPath(container, next) {
+    var opts = {
+      host: process.env.CLOUDANT_HOST,
+      port: process.env.CLOUDANT_PORT,
+      auth: process.env.CLOUDANT_USERNAME + ':' + process.env.CLOUDANT_PASSWORD,
+      path: path,
+    };
+
+    console.log('waiting for instance to respond');
+    return setImmediate(ping, null, 30);
+
+    function ping(err, tries) {
+      console.log('ping (%d/30)', 30 - tries);
+      if (tries < 1) {
+        next(err || new Error('failed to contact Cloudant'));
+      }
+      http.get(opts, function(res) {
+        res.on('error', tryAgain);
+        res.on('data', function() {});
+        res.on('end', function() {
+          if (res.statusCode === 200) {
+            setImmediate(next, null, container);
+          } else {
+            tryAgain();
+          }
+        });
+      }).on('error', tryAgain);
+      function tryAgain(err) {
+        setTimeout(ping, 5000, err, tries - 1);
+      }
+    }
+  };
+}
+
+function createDB(db) {
+  return function create(container, next) {
+    var opts = {
+      method: 'PUT',
+      path: '/' + db,
+      host: process.env.CLOUDANT_HOST,
+      port: process.env.CLOUDANT_PORT,
+      auth: process.env.CLOUDANT_USERNAME + ':' + process.env.CLOUDANT_PASSWORD,
+    };
+    console.log('creating db: %j', db);
+    http.request(opts, function(res) {
+      res.on('data', function() {});
```

Comment:
The no-op `'data'` listener is there to consume the stream so that it ends. I'll try to make that more clear.